source/speechDictHandler/__init__.py

```diff
@@ -118,6 +117,15 @@ def sub(self, text):
 def processText(text):
 	if not globalVars.speechDictionaryProcessing:
 		return text
+	# #14689: API level < 2 UIA consoles have many blank lines, which slows processing to a halt
+	# Because from UIAHandler.constants import WinConsoleAPILevel
+	# cause crashes as global import, and probably slowdowns here
+	# use 2 instead of WinConsoleAPILevel.FORMATTED constant
+	focus = api.getFocusObject()
+	if hasattr(focus, "apiLevel") and focus.apiLevel < 2:
+		stripText = text.rstrip()
+		if len(text) - len(stripText) > 10000:
+			text = stripText

```

Comment:
It would be good to give this constant a name, and make it a bit smaller.
1000 trailing whitespace characters is already a lot. I'd even consider 100.

```suggestion
		IGNORE_TRAILING_WHITESPACE_LENGTH = 1000
		if len(text) - len(stripText) > IGNORE_TRAILING_WHITESPACE_LENGTH:
			text = stripText
```